diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5c6a1ccb..508cafa4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -481,6 +481,14 @@ if(LIEF_COVERAGE)
   target_link_libraries(LIB_LIEF gcov)
 endif()
 
+# Since the Python extension module will be called lief.pyd, we cannot have the C++ Library
+# called LIEF.dll since Windows will not load both.
+if(WIN32 AND BUILD_SHARED_LIBS)
+  set_target_properties(LIB_LIEF PROPERTIES OUTPUT_NAME libLIEF)
+else()
+  set_target_properties(LIB_LIEF PROPERTIES OUTPUT_NAME LIEF)
+endif()
+
 set_target_properties(LIB_LIEF PROPERTIES OUTPUT_NAME LIEF CLEAN_DIRECT_OUTPUT 1)
 
 
