diff -urN a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2021-05-22 11:18:10.000000000 +0300
+++ b/CMakeLists.txt	2022-03-25 17:01:26.940399658 +0300
@@ -481,8 +481,13 @@
   target_link_libraries(LIB_LIEF gcov)
 endif()
 
-set_target_properties(LIB_LIEF PROPERTIES OUTPUT_NAME LIEF CLEAN_DIRECT_OUTPUT 1)
-
+# Since the Python extension module will be called lief.pyd, we cannot have the C++ Library
+# called LIEF.dll since Windows will not load both.
+if(WIN32 AND BUILD_SHARED_LIBS)
+  set_target_properties(LIB_LIEF PROPERTIES OUTPUT_NAME libLIEF)
+else()
+  set_target_properties(LIB_LIEF PROPERTIES OUTPUT_NAME LIEF)
+endif()
 
 # Set a default build type if none was specified
 if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
