diff --git a/src/logging.cpp b/src/logging.cpp
index 3dd9200b..755ac18a 100644
--- a/src/logging.cpp
+++ b/src/logging.cpp
@@ -22,6 +22,16 @@
 #include "spdlog/spdlog.h"
 #include "spdlog/sinks/stdout_color_sinks.h"
 
+
+// Boost infers from the defines/environment that boost::throw_exception is user-defined, so define it here:
+namespace boost {
+void throw_exception(const std::exception& error) {
+  throw error;
+}
+} // namespace boost
+
+
+
 namespace LIEF {
 namespace logging {
 
@@ -147,5 +157,3 @@ void set_level(LOGGING_LEVEL level) {
 
 }
 }
-
-
